tests/test_templates.py

```diff
@@ -12,106 +12,307 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 """
-Unit tests for the integration of the methods from :mod:`pennylane.init` with the :mod:`pennylane.templates` module.
+Integration tests for templates, including integration of passing outputs of initialization functions
+in :mod:`pennylane.init`, and running templates in larger circuits.
 """
 # pylint: disable=protected-access,cell-var-from-loop
 import pytest
+import torch
+import tensorflow as tf
+import numpy as np
 import pennylane as qml
-from pennylane.templates.layers import (CVNeuralNetLayers, CVNeuralNetLayer,
-                                        StronglyEntanglingLayers, StronglyEntanglingLayer,
-                                        RandomLayers, RandomLayer)
-from pennylane.init import (strong_ent_layers_uniform, strong_ent_layer_uniform,
-                            strong_ent_layers_normal, strong_ent_layer_normal,
-                            random_layers_uniform, random_layer_uniform,
-                            random_layers_normal, random_layer_normal,
-                            cvqnn_layers_uniform, cvqnn_layer_uniform,
-                            cvqnn_layers_normal, cvqnn_layer_normal)
+from pennylane.templates.layers import Interferometer
+from pennylane.templates.layers import (CVNeuralNetLayers,
+                                        StronglyEntanglingLayers,
+                                        RandomLayers)
+from pennylane.templates.embeddings import (AmplitudeEmbedding,
+                                            BasisEmbedding,
+                                            AngleEmbedding,
+                                            SqueezingEmbedding,
+                                            DisplacementEmbedding)
+from pennylane.init import (strong_ent_layers_uniform,
+                            strong_ent_layers_normal,
+                            random_layers_uniform,
+                            random_layers_normal,
+                            cvqnn_layers_uniform,
+                            cvqnn_layers_normal)
 
+# When adding a new template,
+# extend the appropriate lists with the template function
+# and the desired inputs (i.e., features and/or weights) that need to be tested
 
-class TestParameterIntegration:
-    """ Integration tests for the parameter generation methods from pennylane.init
-    and pennylane.templates.layers."""
+fixture = "template, inpts"
```

Comment:
This makes things a bit more hidden/hard-to-parse. I think it's fine to put this string in every fixture directly